### ISSUE TITLE
Not able to install Volo.Abp.BackgroundWorkers.Hangfire via ABP CLI

### ISSUE THREAD

**Issue by improwise** [abpframework/abp#11035]
When trying to install abp add-package Volo.Abp.BackgroundWorkers.Hangfire we get this:

```
PS M:\....\src\BackgroundWorkerTest1\aspnet-core\src\BackgroundWorkerTest1.Blazor> abp add-package Volo.Abp.BackgroundWorker.Hangfire
[18:31:37 INF] ABP CLI (https://abp.io)
[18:31:37 INF] Version 5.0.0 (Stable)
[18:31:38 WRN] 'Volo.Abp.BackgroundWorkers.Hangfire' nuget package could not be found!
```
(We tried both with Worker and Workers, see below, same result)

If we however open the project and use Nuget Package manager, it seems like it is there for nuget.org.  If we install it vi NuGet Package Manager and adds:

`typeof(AbpBackgroundWorkersHangfireModule)`

as instructed here:

https://docs.abp.io/en/abp/latest/Background-Workers-Hangfire

we get:

`The type or namespace name 'AbpBackgroundWorkersHangfireModule' could not be found`

If we however change to AbpBackgroundWorkerHangfireModule (Worker instead of Worker**s**)

we can run but instead get 

`InvalidCastException: Unable to cast object of type 'Volo.Abp.Threading.AbpAsyncTimer' to type 'Volo.Abp.Threading.AbpTimer'.`

Not sure if this is just a documentation error (Worker instead of Workers) or if something else goes on that can cause the timer error? Not sure why ABP CLI can't find the package when it is obviously available on NuGet, could it affect that our Solution is created with ABP Suite?

I should add that we are able to run repeating tasks with ABPs default implementation.  Also please note that the documentation for BackgroundWorker does not contain any steps for installing and configure HangFire like in the documentation for BackgroundJobs, like configuring Storage for SQL Server etc. which seems like it would be needed for this to run unless it's handled in memory or something like that. 

Thanks. 

Edit:

I should add that "abp add-package Volo.Abp.BackgroundJobs.HangFire" works fine, so it seems like it is just the abp add-package Volo.Abp.BackgroundWorkers.Hangfire  (Workers instead of Jobs) that does not work with abp add-package.

**Comment by improwise** [comment]
"abp add-package Volo.Abp.BackgroundWorkers.Quartz" also works fine and installs the package as it should.

**Comment by rqx110** [comment]
it should "Volo.Abp.BackgroundWorker**s**.Hangfire", you have missed "s" in your command

**Comment by improwise** [comment]
> it should "Volo.Abp.BackgroundWorker**s**.Hangfire", you have missed "s" in your command

Thanks, but please read the issue  more in detail, we already tried both. 

Log file from CLI just to show that this problem seems to be specific to this package:


```
2021-12-22 23:33:40.396 +01:00 [INF] ABP CLI (https://abp.io)
2021-12-22 23:33:40.580 +01:00 [INF] Version 5.0.0 (Stable)
2021-12-22 23:33:41.623 +01:00 [WRN] 'Volo.Abp.BackgroundWorkers.Hangfire' nuget package could not be found!
2021-12-22 23:34:07.958 +01:00 [INF] ABP CLI (https://abp.io)
2021-12-22 23:34:08.145 +01:00 [INF] Version 5.0.0 (Stable)
2021-12-22 23:34:08.902 +01:00 [INF] Installing 'Volo.Abp.BackgroundWorkers.Quartz' package to the project 'BackgroundWorkerTest2.Blazor'...
2021-12-22 23:34:17.684 +01:00 [INF] Successfully installed.
2021-12-22 23:37:59.287 +01:00 [INF] ABP CLI (https://abp.io)
2021-12-22 23:37:59.477 +01:00 [INF] Version 5.0.0 (Stable)
2021-12-22 23:38:00.863 +01:00 [WRN] 'Volo.Abp.BackgroundWorkers.Hangfire' nuget package could not be found!
```

**Comment by realLiangshiwei** [comment]
Try `dotnet add package Volo.Abp.BackgroundWorkers.Hangfire --version 5.0.0`

**Comment by improwise** [comment]
> Try `dotnet add package Volo.Abp.BackgroundWorkers.Hangfire --version 5.0.0`

That works as well as doing it in the NuGet Package Manager, but we are thinking it must be a reson for this specific problem with the CLI and as mentioned above, we have other problems after having added it another way. Please also note that the Depends on does not match with what's in the docs (which could be a typo). All in all it seems a bit weird and we can't get it to work all the way. Works with just the ABP Worker though as well as with HangFire Background jobs.

**Comment by realLiangshiwei** [comment]
We will check it.

**Comment by Nokecy** [comment]
https://github.com/abpframework/abp/blob/4b87fc8ae5b4e1637b16553ffea7acdff2d96ec5/framework/src/Volo.Abp.BackgroundWorkers.Hangfire/Volo/Abp/BackgroundWorkers/Hangfire/HangfireBackgroundWorkerManager.cs#L45

AsyncPeriodicBackgroundWorkerBase  use AbpAsyncTimer
@realLiangshiwei

**Comment by realLiangshiwei** [comment]
@Nokecy Thanks,  see https://github.com/abpframework/abp/pull/11061

**Comment by improwise** [comment]
Good that a solution seem to be available. Thanks. 

This probably explains the crash we got when we configured this manually. But any idea of why the nuget wasn't possible to install via ABP CLI? Also note that there was either a small typo in the docs or the DependsOn statement had lost an "s" in Workers as mentioned in the first post. 

Do we need to switch our entire SLN to Nightly builds to get hold of this fix before it is included in an actual release or can we just manually install only this package as a nightly build?
